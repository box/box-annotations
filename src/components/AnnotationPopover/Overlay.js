// @flow
import * as React from 'react';

import OverlayComponent from 'box-react-ui/lib/components/flyout/Overlay';

const CLASS_POPOVER_OVERLAY = 'ba-popover-overlay';

type Props = {
    shouldDefaultFocus: boolean,
    children: React.Node
};

const Overlay = ({ shouldDefaultFocus, children }: Props) => {
    if (shouldDefaultFocus) {
        return (
            <OverlayComponent className={CLASS_POPOVER_OVERLAY} shouldOutlineFocus={false}>
                {children}
            </OverlayComponent>
        );
    }

    return <div className={CLASS_POPOVER_OVERLAY}>{children}</div>;
};

export default Overlay;
