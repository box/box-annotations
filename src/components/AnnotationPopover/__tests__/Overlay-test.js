import * as React from 'react';
import { shallow } from 'enzyme';

import Overlay from '../Overlay';

describe('components/Overlay', () => {
    test('should correctly render a Focus Trap if not on mobile', () => {
        const wrapper = shallow(<Overlay shouldDefaultFocus={true} />).dive();
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('FocusTrap').length).toEqual(1);
    });

    test('should correctly render a div without a Focus Trap if on mobile', () => {
        const wrapper = shallow(<Overlay shouldDefaultFocus={false} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('FocusTrap').length).toEqual(0);
    });
});
