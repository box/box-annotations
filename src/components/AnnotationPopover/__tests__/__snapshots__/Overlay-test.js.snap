// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Overlay should correctly render a Focus Trap if not on mobile 1`] = `
<FocusTrap
  className="ba-popover-overlay"
  handleOverlayKeyDown={[Function]}
  shouldOutlineFocus={false}
  tabIndex={0}
>
  <div
    className="overlay"
  />
</FocusTrap>
`;

exports[`components/Overlay should correctly render a div without a Focus Trap if on mobile 1`] = `
<div
  className="ba-popover-overlay"
/>
`;
